src/core/stream/video_events.ts

```diff
@@ -59,37 +61,47 @@ function doInitialSeek(
 }
 
 /**
- * @param {HTMLMediaElement} videoElement
- * @param {number|Function} startTime
- * @param {boolean} autoPlay
+ * @param {HTMLMediaElement} mediaElement
+ * @param {number|Function} startTime - Initial starting position. As seconds
+ * or as a function returning seconds.
+ * @param {boolean} autoPlay - Whether the player should auto-play
  * @returns {object}
  */
-export default function handleVideoEvents(
-  videoElement : HTMLMediaElement,
+export default function seekAndLoadOnMediaEvent(
+  mediaElement : HTMLMediaElement,
   startTime : number|(() => number),
   mustAutoPlay : boolean
 ) : {
-  initialSeek$ : Observable<void>;
-  loadAndPlay$ : Observable<void>;
+  seek$ : Observable<void>;
+  load$ : Observable<"autoplay-blocked"|"autoplay"|"loaded">;
 } {
-  const initialSeek$ = doInitialSeek(videoElement, startTime);
-  const handledCanPlay$ = canPlay(videoElement).pipe(
-    tap(() => log.info("canplay event"))
-  );
+  const seek$ = doInitialSeek(mediaElement, startTime);
+
+  const load$ = canPlay(mediaElement).pipe(
+
+    tap(() => log.info("canplay event")),
+
+    mergeMap(() => {
+      if (mustAutoPlay) {
+        return play$(mediaElement).pipe(
+          mapTo("autoplay" as "autoplay"),
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove this unnecessary cast. ([typescript:S4325](https://sonarcloud.io/organizations/rx-player/rules#rule_key=typescript%3AS4325))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=rx-player&pullRequest=282&issues=AWUkWYcQOTG07Bqb8h4k&open=AWUkWYcQOTG07Bqb8h4k)

[ISSUE_KEY]: AWUkWYcQOTG07Bqb8h4k